gef.py

```diff
@@ -9181,6 +9181,125 @@ def get_settings_path(self):
         path = os.path.realpath(path)
         return path if os.path.isdir(path) else None
 
+@register_command
+class BincompareCommand(GenericCommand):
+    """BincompareCommand: compare an binary file with the memory position looking for badchars."""
+    _cmdline_ = "bincompare"
+    _syntax_  = "{:s} -f FILE -a MEMORY_ADDRESS [-h]".format(_cmdline_)
+
+    def __init__(self):
+        super(BincompareCommand, self).__init__(complete=gdb.COMPLETE_FILENAME)
+        return
+
+
+    def usage(self):
+        h = self._syntax_
+        h += "\n\t-f FILE specifies the binary file to be compared.\n"
+        h += "\t-a MEMORY_ADDRESS sepecifies the memory address.\n"
+        info(h)
+        return
+
+    @only_if_gdb_running
+    def do_invoke(self, argv):
+        filename = None
+        start_addr = None
+        size = 0
+        file_data = None
+        memory_data = None
+
+        opts, args = getopt.getopt(argv, "f:a:ch")
+        for o,a in opts:
+            if   o == "-f": filename = a
+            elif o == "-a": start_addr = int(gdb.parse_and_eval(a))
+            elif o == "-h":
```

Comment:
`-h` is unnecessary, the docstring of the command and the `usage` build the help menu if you call the GDB `help <command>`